### PR TITLE
Adhere to new "resolve" field when resolving video url

### PR DESCRIPTION
**Pull request by even-stephen**
A new field "resolve" has been introduced in the video response from the svt api, this field
should be preferred over the alt query parameter since it will most likely be removed in the
near future.

### CODE REVIEW

File: resources/lib/helper.py [hunk 1]
```diff
@@ -72,24 +73,21 @@ def getVideoURL(json_obj):
...
   video_url = None
   for video in json_obj["videoReferences"]:
-    if video["format"] == "hls":
-      alt = getAltUrl(video["url"])
-      if alt is None:
-        video_url = video["url"]
-      else:
-        video_url = alt
+    if video["format"] == "hls":            
+      if 'resolve' in video:
```
Comment (linqcan) [inline resources/lib/helper.py]: Use double quotes. Please check all your changes.

File: resources/lib/helper.py [hunk 2]
```diff
@@ -72,24 +73,21 @@ def getVideoURL(json_obj):
...
+         video_url = getResolvedUrl(video['resolve'])
+      if video_url is None:
+          video_url = video["url"]        
   return video_url
 
-def getAltUrl(video_url):
-  o = urlparse(video_url)
```
Comment (linqcan) [inline resources/lib/helper.py:84]: If we don't need these methods from url lib, can you clean up the import at the top?

**Review by linqcan — CHANGES_REQUESTED** [review]
Hello again! 👋 

Could you remind me, why is this needed today?

Code looks fine except some minor comments that needs fixing.

File: resources/lib/helper.py (hunk 1 above)
Comment (even-stephen) [reply resources/lib/helper.py]: good point, fixed

File: resources/lib/helper.py (hunk 2 above)
Comment (even-stephen) [reply resources/lib/helper.py:84]: good point, fixed

**Comment by even-stephen** [comment]
Clarification: the resolve field should be considered as a "alt-query" replacement, i.e it provides the same functionallty in terms of correctly resolving a video url but without needing to follow HTTP redirects(and is future proof).

**Comment by linqcan** [comment]
> Clarification: the resolve field should be considered as a "alt-query" replacement, i.e it provides the same functionallty in terms of correctly resolving a video url but without needing to follow HTTP redirects(and is future proof).

Could you clarify what the functionality is?
The old commit states that some devices need this. Which ones? Why? and Do they still today?

**Comment by even-stephen** [comment]
"Old devices" where not capable of following the redirect(s) which would come from following the alt link. The resolve link can be used regardless of devices being old or new (since there is there is no HTTP redirect involved) The functionality of the resolve link is the same as the alt link (better CDN selection)

**Comment by linqcan** [comment]
But what are "old devices" (examples?) and why are they relevant?

**Comment by even-stephen** [comment]
Old devices = devices running old versions of kodi/ffmpeg that are incapable of following redirects properly. Ex: https://forum.kodi.tv/showthread.php?tid=194844

Supporting "old devices" is not an issue at all in this context (since there is no more redirect.)

**Comment by linqcan** [comment]
Thanks for clarifying! 👍🏻

If you could squash the commits to one, then I can merge it. 

Thanks for contributing!

**Comment by even-stephen** [comment]
done

**Pull request merged**